VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/FileRequest.tsx

```diff
@@ -38,18 +38,31 @@ const FileRequest: FC<FileRequestProps> = ({ navigation, route }) => {
       navigateTo('FileRequestDetails', { claimID, request })()
     }
 
-    return map(requests, (request) => {
+    const getA11yLabel = (requestIndex: number, displayName?: string, uploaded?: boolean) => {
+      const nameContainsRequestNumber = displayName && /Request \d/.test(displayName)
+      const status = uploaded ? t('uploaded') : t('needed')
+
+      if (!displayName) {
+        return `${t('fileRequest.buttonA11y', { requestNumber: requestIndex, totalCount: requests.length })} ${status}`
+      }
+
+      return nameContainsRequestNumber
+        ? `${displayName} ${t('fileRequest.buttonA11y.totalCount', { totalCount: requests.length })} ${status}`
+        : `${t('fileRequest.buttonA11y', { requestNumber: requestIndex, totalCount: requests.length })} ${status}. ${displayName}`
+    }
+
+    return map(requests, (request, index) => {
       const { displayName } = request
       const hasUploaded = hasUploadedOrReceived(request)
       const item: SimpleListItemObj = {
         text: displayName || '',
-        testId: displayName,
+        testId: getA11yLabel(index + 1, displayName, hasUploaded),
```

Comment:
Would it make sense to use this as an opportunity to separate out the testId from the a11ylabel?

Reply:
Good question, initially I looked into that, but `testId` is the only place where we can pass an accessibility label, since the `BaseListItem` props don't include `accessibilityLabel`. We'd need to update the `BaseListItem` component to accept and use that prop, which might be getting out of scope since it could impact other parts of the app using lists. I'm not entirely sure, but this issue may get addressed in the work for https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6646